editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -206,6 +206,20 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
   }
 };
 
+/**
+ * Generate meaningful title for content analytic log
+ */
+function getContentTitle(node: Store.node): string {
+  const dom = new DOMParser().parseFromString(node.data.content, "text/html");
+  const text = dom.body.textContent;
+  if (!text) return `Content: ${node.id}`;
+  const TITLE_LENGTH = 50;
+  const truncate = (data: string) =>
+    data.length > TITLE_LENGTH ? data.substring(0, TITLE_LENGTH) + "..." : data;
```

Comment:
I actually never thought of this...!

There's nothing actually enforcing the requirement for `<h1>` tags currently so the better solution here might be to add a title field to content eventually?

For now I've added a check so the logic is basically - 
 - Use `<h1>` as title
 - Otherwise fallback to first 50 characters of `textContent`